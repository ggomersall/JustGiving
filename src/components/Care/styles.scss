@import '../BaseStyles/variables';

.care__wrapper {
  display: flex;
  font-size: 20px;
  .care__button button{
    border: 1px solid $darkgrey;
    height: 40px;
    width: 100px;
    font-size: 20px;
    &:hover {
      border-color: $purple;
      color: $purple;
    }
    &.cared {
      background: $purple;
      color: $white;
    }
  }

  .care__totals span{
    border: 1px solid $darkgrey;
    display: block;
    height: 38px;
    line-height: 38px;
    padding: 0 20px;
    width: auto;
    margin-left: 15px;
  }
}