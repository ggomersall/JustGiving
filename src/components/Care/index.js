import React from 'react';
import classNames from 'classnames';
import './styles.scss';

const Care = ({likes, handleCareUpdate, care}) => {
  return (
    <div className="care__wrapper">
      <div className="care__button">
        <button className={classNames({'cared': care != false})} onClick={handleCareUpdate}>{care === false ? 'Care' : 'I care'}</button>
      </div>
      <div className="care__totals">
        <span>{likes}</span>
      </div>
    </div>
  )
}

export default Care;