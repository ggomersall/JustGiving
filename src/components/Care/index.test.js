import React from 'react';
import { shallow } from 'enzyme';
import Care from './index.js';

describe('Testing Care Component', () => {

  let wrapper;

  it('renders correctly', () => {
    shallow(<Care />);
  });

  test('it should render likes and care props correctly', () => {
    const likes = 100;
    const care = true;
    wrapper = shallow(<Care likes={likes} care={care} />);
  });

  test('Snapshot', () => expect(<Care />).toMatchSnapshot());

});

