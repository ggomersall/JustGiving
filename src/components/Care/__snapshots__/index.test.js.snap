// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Care Component Snapshot 1`] = `<Care />`;
